### PR TITLE
Fluid bytecode c++ executor.

### PR DESCRIPTION
**Pull request by gongweibao**
Fix https://github.com/PaddlePaddle/Paddle/issues/10221

### CODE REVIEW

File: python/paddle/fluid/executor.py [hunk 1]
```diff
@@ -333,6 +335,12 @@ def run(self,
                 fetch_var_name=fetch_var_name)
 
         self._feed_data(program, feed, feed_var_name, scope)
+
+        # TODO(gongwb): does a program should be saved in run function?
+        if len(save_program_to_file) > 0:
```
[the remark this answers is not in the document]
Comment (gongweibao) [reply python/paddle/fluid/executor.py]: The programdesc is modified in run function.And sometimes a user can't get chance to save it.

**Comment by typhoonzero** [comment]
Agree to put this to `tools`

File: paddle/fluid/operators/detail/desc_executor.cc [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  framework::proto::ProgramDesc proto;
+  if (!proto.ParseFromArray(buf, buf_len)) {
+    printf("parse from %s error!\n", file.c_str());
+    return NULL;
+  }
+
+  return (new framework::ProgramDesc(proto));
+}
+
+DEFINE_string(start_up_proto, "", "start up proto file");
```
[the remark this answers is not in the document]
Comment (gongweibao) [reply paddle/fluid/operators/detail/desc_executor.cc]: Thanks.Done.

File: paddle/fluid/operators/detail/desc_executor.cc [hunk 3]
```diff
@@ -0,0 +1,126 @@
+//   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (gongweibao) [reply paddle/fluid/operators/detail/desc_executor.cc]: Done.

File: paddle/fluid/operators/detail/desc_executor.cc [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+#include "paddle/fluid/framework/init.h"
+#include "paddle/fluid/framework/op_registry.h"
+#include "paddle/fluid/framework/program_desc.h"
+#include "paddle/fluid/framework/scope.h"
+#include "paddle/fluid/pybind/pybind.h"
+
+bool read_from_file(const std::string& file, char** buf, int64_t* buf_len) {
+  FILE* f = fopen(file.c_str(), "rb");
+  if (NULL == f) {
+    printf("open %s error\n", file.c_str());
```
[the remark this answers is not in the document]
Comment (gongweibao) [reply paddle/fluid/operators/detail/desc_executor.cc]: change to `fprintf(stderr`.
Thanks.

File: paddle/fluid/operators/detail/desc_executor.cc [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  *buf_len = fsize;
+
+  (*buf)[fsize] = 0;
+  fclose(f);
+  return true;
+}
+
+using namespace paddle;  // NOLINT
+
+framework::ProgramDesc* load_desc(const std::string& file) {
```
[the remark this answers is not in the document]
Comment (gongweibao) [reply paddle/fluid/operators/detail/desc_executor.cc]: Done.

File: python/paddle/fluid/executor.py [hunk 6]
```diff
@@ -277,7 +277,8 @@ def run(self,
             fetch_var_name='fetch',
             scope=None,
             return_numpy=True,
-            use_program_cache=False):
+            use_program_cache=False,
+            save_program_to_file=""):
```
Comment (panyx0718) [inline python/paddle/fluid/executor.py]: I still don't want to put "save_program_to_file" as a argument to Executor. Saving program should be in a separate API, such as "save_inference_model".

File: python/paddle/fluid/executor.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (panyx0718) [reply python/paddle/fluid/executor.py]: Maybe run() shouldn't modify the programdesc? Or we should save after run is called?

**Comment by gongweibao** [comment]
I compile debug-version Python and so close this PR.

**Pull request closed without merging**